packages/editor/src/editor.ts

```diff
@@ -0,0 +1,236 @@
+      editor = process.env.EDITOR;
+    }
+
+    return editor;
+  }
+
+  async _createTempFile(): Promise<void> {
+    const ext = await this._getExtension(this._editorSession.cmd);
+    const content = await this._getEditorContent();
+
+    this._editorSession.tmpDoc = path.join(this._tmpDir, `edit-${uuidv4()}.${ext}`);
+
+    // Create a temp file to store content that is being edited.
+    await fse.ensureFile(this._editorSession.tmpDoc);
+    await fse.writeFile(this._editorSession.tmpDoc, beautify(content));
+  }
+
+  async _readTempFile(tmpDoc: string): Promise<string> {
+    const content = await fse.readFile(tmpDoc, 'utf8');
+    await fse.unlink(tmpDoc);
+    return this._makeMultilineJSIntoSingleLine(content);
+  }
+
+  _isVscodeApp(cmd: string): boolean {
+    const regex = /^((.*[\/\\])*)?[cC]ode(.exe)?$/;
+
+    try {
+      return !!regex.exec(cmd);
+    } catch (error) {
+      this.messageBus.emit('mongosh-editor:is-vscode-app-regex-failed', { regex, cmd });
+      return false;
+    }
+  }
+
+  _isStatement(code: string): boolean {
+    const regex = /\b[^()]+\((.*)\)$/;
+
+    try {
+      return !!regex.exec(code);
+    } catch (error) {
+      this.messageBus.emit('mongosh-editor:is-statement-regex-failed', { regex, code });
+      return false;
+    }
+  }
+
+  async _getEditorContent(): Promise<string> {
+    if (!this._editorSession.code) {
+      return this._editorSession.content;
+    }
+
+    const evalResult = await this._loadExternalCode(this._editorSession.code, '@(shell eval)');
+    return this._isStatement(this._editorSession.code) ? this._editorSession.code : evalResult.toString();
+  }
+
+  async runEditCommand([ code, ...args ]: string[]): Promise<void> {
+    await this.print('Opening an editor...');
+    const editor: string|null = await this._getEditor();
+
+    // If none of the above configurations are found return an error.
+    if (!editor) {
+      throw new Error(`Command failed: ${[code, ...args].join(' ')} with an error: please define an external editor`);
+    }
+
+    const [ cmd, ...cmdArgs ] = editor.split(' ');
+
+    this._editorSession.cmd = cmd;
+    this._editorSession.args = [ ...args, ...cmdArgs ];
+    this._editorSession.code = code;
+
+    this.messageBus.emit('mongosh-editor:run-edit-command', {
+      tmpdoc: this._editorSession.tmpDoc,
+      editor,
+      args: this._editorSession.args
+    });
+
+    await this._createTempFile();
+    await this._runInChildProcess();
+  }
+
+  async _runInChildProcess(): Promise<void> {
+    const { cmd, tmpDoc, args, code } = this._editorSession;
+    const proc = spawn(cmd, [tmpDoc, ...args], {
+      env: { ...process.env, MONGOSH_RUN_NODE_SCRIPT: '1' },
+      stdio: 'inherit'
+    });
+
+    let stdout = '';
+    let stderr = '';
+
+    if (proc.stdout) {
+      proc.stdout.on('data', (chunk) => { stdout += chunk; });
+    }
+    if (proc.stderr) {
+      proc.stderr.on('data', (chunk) => { stderr += chunk; });
+    }
```

Comment:
```suggestion
    (proc.stdout as NodeJS.ReadableStream).setEncoding('utf8').on('data', (chunk) => { stdout += chunk; });
    (proc.stderr as NodeJS.ReadableStream).setEncoding('utf8').on('data', (chunk) => { stderr += chunk; });
```

matches what snippet-manager does. (The `.setEncoding('utf8')` call is missing in any case.)